lib/SILOptimizer/Mandatory/DIMemoryUseCollectorOwnership.cpp

```diff
@@ -1115,10 +1071,26 @@ void ElementUseCollector::collectClassSelfUses() {
   for (auto *Op : MUI->getUses()) {
     SILInstruction *User = Op->getUser();
 
-    // Stores to self are initializations store or the rebind of self as
-    // part of the super.init call.  Ignore both of these.
-    if (isa<StoreInst>(User) && Op->getOperandNumber() == 1)
-      continue;
+    // Stores to self.
+    if (auto *SI = dyn_cast<StoreInst>(User)) {
+      if (Op->getOperandNumber() == 1) {
+        // The initial store of 'self' into the box at the start of the
+        // function. Ignore it.
+        if (auto *Arg = dyn_cast<SILArgument>(SI->getSrc()))
```

Comment:
Oops, I didn't update this singular comment. This is a case where you should comment that you know that the argument is owned and can only be consumed once since otherwise the ownership verifier would blow up. That being said, you should probably add an assert that you only see one of these while iterating over the loop.